sdks/java/testing/jpms-tests/build.gradle

```diff
@@ -21,7 +21,14 @@ import groovy.json.JsonOutput
 plugins {
   id 'org.apache.beam.module'
 }
-javaVersion="1.11"
+
+// overwrite javaVersion before applyJavaNature
+if (project.hasProperty("compileAndRunTestsWithJava17")) {
+  javaVersion = '1.17'
+} else {
+  javaVersion = '1.11'
```

Comment:
this is subtle. Current Java11/Java17 test are configured differently.

- Java11 passed `-Dorg.gradle.java.home=$JAVA_HOME_11_X64` so all projects (e.g. java core) are also compared with Java11

- Java17 uses `compileAndRunTestsWithJava17` flag so all but this project are compiled with default JDK (Java8)

Reply:
Thanks, noted in https://github.com/apache/beam/issues/28120#issuecomment-1741033265 will be done as part of Java 21 support